core/include/kernel/dt.h

```diff
@@ -199,10 +199,10 @@ static inline paddr_t _fdt_reg_base_address(const void *fdt __unused,
 	return (paddr_t)-1;
 }
 
-static inline ssize_t _fdt_reg_size(const void *fdt __unused,
-				    int offs __unused)
+static inline size_t _fdt_reg_size(const void *fdt __unused,
+				   int offs __unused)
 {
-	return -1;
+	return (size_t)-1;
```

Comment:
prefer `return DT_INFO_INVALID_REG_SIZE;`

Reply:
Idea here is to be compliant with the previous function that return (paddr_t)-1 too instead of DT_INFO_INVALID_REG

Reply:
sorry, my comment does not apply to your P-R.
You should  have another patch to update all these casted `-1` to their related `DT_INFO_INVALID_*`, IMO.